pkg/destroy/azure/azure.go

```diff
@@ -602,6 +602,10 @@ func deleteApplicationRegistrations(ctx context.Context, graphClient *msgraphsdk
 			errorList = append(errorList, err)
 			continue
 		}
+		// msgraphsdk can return a `nil` response even if no errors occurred
+		if resp == nil {
+			continue
+		}
```

Comment:
Do we know what a nil response means? Was the deletion successful? API not supported? 

If we don't know, or aren't certain, I would suggest do a little more here in terms of logging, perhaps a warning or or adding an error to the error list.

Reply:
In this case it's just a GET (all the apps with ID == `appID`), the delete happens a few lines later. I cannot easily tell why, would have to dig dipper into the msgraph code: https://github.com/microsoftgraph/msgraph-sdk-go/blob/main/serviceprincipals/service_principal_item_request_builder.go#L267-L273

Reply:
I think this is the `SendAsync` function: https://github.com/microsoft/kiota-http-go/blob/main/nethttp_request_adapter.go#L286-L287

Reply:
Anyway, it's probably a good idea to at least add some logging in those cases. Thanks.